gems/smithy/lib/smithy/generators/client.rb

```diff
@@ -94,6 +95,12 @@ def code_generated_plugins
       def should_skip_customizations?
         Dir["#{destination_root}/**/*"].any? { |f| f.include?('/customizations.rb') }
       end
+
+      def protocol_tests?
```

Comment:
I'm assuming you did this to avoid generating an empty file. I don't like knowledge of these traits in the orchestrator. I would be ok with generating an empty file with maybe a single default test. We should smoke test at least something.

Reply:
It feels a little weird to generate an empty protocol test file for most gems.   We could move the trait knowledge to some utility class.

Reply:
Is there not a default test we could have, like test the option works?

Reply:
The `protocol` configuation option you mean?

That seems like a different test - a `protocol_plugin_spec`.

Reply:
I don't think a utility class makes sense because it's just hiding away the same problem and only used once. I think a file with no tests is reasonable or we can keep this as is